### DISCUSSION TITLE
remark-lint-list-item-style exceptions

### DISCUSSION THREAD

**Discussion by fviolette** [Xunnamius/unified-utils#47] [Q&A] [answered]
It occurs that some list items can start with a lowercase letter such as `- iOS:`.
In this case, applying the `checkFirstWord` option will raise a warning.

I was wondering if a configurable pattern could be used to bypass this, let's say `- **iOS**:`. It would be our responsibility to enforce it then in our writing convention.
Ignoring the rule for such lines looks overkill to me right now, but I'd like to use the package across the rest of the docs.

Quite the custom ask but this has happened in other projects for sure.

Thanks for your feedback.

**Comment by Xunnamius** [comment, marked as answer]
Sounds like a worthy feature. Perhaps an `ignoredFirstWords` option that accepts an array of strings/RegExps that causes all matching first words to be ignored.

If that sounds good, I'll demo it now. In the interim, you can also use [remark's lint ignore command](https://github.com/remarkjs/remark-lint#ignore-warnings) to suppress the warning only for that line (not sure if this is what you were referring to in your second paragraph).

UPDATE: Try version [1.1.0](https://github.com/Xunnamius/unified-utils/releases/tag/remark-lint-list-item-style%401.1.0).

**Reply by fviolette** [reply]
Great, that's quick — I wasn't able to make this work yet.
Adding `"ignoredFirstWords": [/^n/, '^bar', 'iOS']` to the same config block shared in https://github.com/Xunnamius/unified-utils/issues/43.

With the markup below iOS is still flagged as inconsistent:

```
## 1.0

- iOS SDK first version
- Android SDK first version
```

**Reply by Xunnamius** [reply]
Oops, I forgot to trigger the rebuild first (my pub process is a little messy while I switch things over to a new system). It should be working now. You don't have to include the first two unrelated regular expressions, either.

Try version [1.1.1](https://github.com/Xunnamius/unified-utils/releases/tag/remark-lint-list-item-style%401.1.1).

**Comment by fviolette** [comment]
It does work now, thank you ✅
